### PR TITLE
Feature/collaborators

### LINKED ISSUE #31: As a signed in user I can invite other users to my account as collaborators
In my account, I can go to a collaborators page and invite a collaborator via email. I can see a list of collaborators, including their names (if they already accepted the invitation), the email and when I invited them. In the future we want to limit how many collaborators a user is allowed to add, for now, there is no limit.

- Add [devise_invitable](https://github.com/scambra/devise_invitable) gem
- Add index page for collaborators
- Update new invite view to match our theme
- Update invitation view to match our theme
- We create the invited user's account with a random name
- we assign the invited user to account of the user who sent the invitation

### PR DESCRIPTION
**Pull request by philip13**
Close #31

Screen Shot 2023-04-10 at 14 47 10
Screen Shot 2023-04-10 at 14 47 24
Screen Shot 2023-04-10 at 14 48 15
Screen Shot 2023-04-10 at 14 47 59

### CODE REVIEW

File: app/controllers/collaborators_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class CollaboratorsController < ApplicationController
+  def index
+    @collaborators = current_user.collaborators
+  end
+
+  def resend_invitation
```
Comment (ingilniero) [inline app/controllers/collaborators_controller.rb:6]: Move this action to its own controller (e.g `InvititationResendsController#create`, we only want to use the RESTful actions.

File: app/helpers/status_tag_helper.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module StatusTagHelper
+  def status_tag(active)
+    if active
+      "<span class='inline-flex px-2 text-xs font-semibold leading-5 text-green-800 bg-green-100 rounded-full'>
+        #{t("yesno.yes")}
```
Comment (ingilniero) [inline app/helpers/status_tag_helper.rb:5]: change this to be `active` or `inactive`

File: app/helpers/status_tag_helper.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module StatusTagHelper
+  def status_tag(active)
+    if active
+      "<span class='inline-flex px-2 text-xs font-semibold leading-5 text-green-800 bg-green-100 rounded-full'>
```
Comment (ingilniero) [inline app/helpers/status_tag_helper.rb:4]: interpolate only the css class that changes, you could use the `token_list/class_names` helper might come handy

File: app/views/collaborators/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        <div class="flex px-4 py-4 space-x-4 overflow-x-auto bg-white rounded-md">
+          <%= link_to t(".invite"), new_user_invitation_path, class: "px-4 py-2 font-medium tracking-wide text-white capitalize transition-colors duration-200 transform bg-indigo-600 rounded-md hover:bg-indigo-500 focus:outline-none focus:bg-indigo-500" %>
+        </div>
+
+        <div class="inline-block min-w-full overflow-hidden align-middle mt-4 border-b border-gray-200 shadow sm:rounded-lg" >
+          <table class="min-w-full">
+            <thead class="border-b">
+              <tr>
+                <th class="px-5 py-3 text-sm font-medium text-left text-gray-100 uppercase bg-indigo-800">
+                  Name
```
Comment (ingilniero) [inline app/views/collaborators/index.html.erb:17]: Add translations for the column names

File: app/views/collaborators/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                </th>
+                <th class="px-5 py-3 text-sm font-medium text-left text-gray-100 uppercase bg-indigo-800">
+                  Email
+                </th>
+                <th class="px-5 py-3 text-sm font-medium text-left text-gray-100 uppercase bg-indigo-800">
+                  Confirmed?
```
Comment (ingilniero) [inline app/views/collaborators/index.html.erb:23]: Use `Status` for this column

File: app/views/shared/_sidebar.html.erb [hunk 6]
```diff
@@ -20,6 +20,11 @@
         <%= svg("clients", class: "w-5 h-5") %>
         <span class="mx-4">Clients</span>
       <% end %>
+
+      <%= link_to collaborators_path, class: link_class(collaborators_path) do %>
+        <%= svg("clients", class: "w-5 h-5") %>
+        <span class="mx-4">Collaborators</span>
```
Comment (ingilniero) [inline app/views/shared/_sidebar.html.erb:26]: Add a translation for this text

File: test/system/collaborator_accept_invitation.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+class CollaboratorAcceptInvitation < ApplicationSystemTestCase
+  test "create user account" do
+    collaborator = users(:collaborator1)
+    collaborator.invite!
+    token = collaborator.raw_invitation_token
+
+    visit accept_user_invitation_url(invitation_token: token)
+
+    fill_in "Name", with: collaborator.name
```
Comment (ingilniero) [inline test/system/collaborator_accept_invitation.rb:11]: Use `I18n.human_attribute_for(:name)` translation for the name of the fields

**Review by ingilniero — CHANGES_REQUESTED** [review]

File: test/system/collaborator_accept_invitation.rb (hunk 7 above)
Comment (philip13) [reply test/system/collaborator_accept_invitation.rb:11]: We can't use `I18n.human_attribute_for(:name)` 
but works with 
`Client.human_attribute_name(:name)`

Comment (ingilniero) [reply test/system/collaborator_accept_invitation.rb:11]: ah, that's the helper I was referring to, yes use that

File: app/helpers/status_tag_helper.rb [hunk 8]
```diff
@@ -1,13 +1,12 @@
...
-      </span>".html_safe
-    else
-      "<span class='inline-flex px-2 text-xs font-semibold leading-5 text-red-800 bg-red-100 rounded-full'>
-        #{t("yesno.no")}
-      </span>".html_safe
-    end
+  def t_active(active)
+    active ? t("active") : t("inactive")
+  end
+
```
Comment (ingilniero) [inline app/helpers/status_tag_helper.rb:5]: remove this extra space

File: app/views/collaborators/index.html.erb [hunk 9]
```diff
@@ -14,13 +14,13 @@
             <thead class="border-b">
               <tr>
                 <th class="px-5 py-3 text-sm font-medium text-left text-gray-100 uppercase bg-indigo-800">
-                  Name
+                  <%= t("users.attributes.name") %>
                 </th>
                 <th class="px-5 py-3 text-sm font-medium text-left text-gray-100 uppercase bg-indigo-800">
-                  Email
+                  <%= t("users.attributes.email") %>
```
Comment (ingilniero) [inline app/views/collaborators/index.html.erb:20]: same here

File: app/views/shared/_sidebar.html.erb [hunk 10]
```diff
@@ -1,4 +1,4 @@
-<div class="flex">
+  <div class="flex">
```
Comment (ingilniero) [inline app/views/shared/_sidebar.html.erb:1]: remove extra space

File: config/locales/globals/en.yml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 en:
   app_name: AgroReports
   dashboard: Dashboard
-  yesno:
-    yes: "Yes"
-    no: "No"
+  active: Active
```
Comment (ingilniero) [inline config/locales/globals/en.yml:4]: this shouls be in `config/locales/views/collaborators/en.yml`

File: app/controllers/invititation_resends_controller.rb [hunk 12]
```diff
@@ -0,0 +1,12 @@
+class InvititationResendsController < ApplicationController
```
Comment (ingilniero) [inline app/controllers/invititation_resends_controller.rb:1]: can you add a test for this controller?

**Review by ingilniero — CHANGES_REQUESTED** [review]

**Review by ingilniero — APPROVED** [review]

**Pull request merged**